pkg/model/v1/segment_decoder.go

```diff
@@ -23,9 +23,9 @@ func (d *SegmentDecoder) PrepareForWrite(trace *tempopb.Trace, _, _ uint32) ([]b
 	return proto.Marshal(trace)
 }
 
-func (d *SegmentDecoder) PrepareForRead(segments [][]byte) (*tempopb.Trace, error) {
+func (d *SegmentDecoder) PrepareForRead(segments [][]byte, allowPartialTraces bool) (*tempopb.Trace, error) {
 	// each slice is a marshalled tempopb.Trace, unmarshal and combine
-	combiner := trace.NewCombiner(0)
+	combiner := trace.NewCombiner(0, allowPartialTraces)
```

Comment:
let's revert the changes to `PrepareForRead`. We shouldn't need any changes here. just hardcode `false` or `true`. It shouldn't matter since `0` means no limit.

Reply:
Ahh that's why my comment here: https://github.com/grafana/tempo/blob/26721832e2b1320de39183338d0d58d9da4083a6/pkg/model/trace/combine.go#L152

From the tests, I wasn't sure of the purpose of that, maybe we should document it or adding something more explicit like checkSize bool